### PR TITLE
Homework21: Andrei Kulbitski

### PR DESCRIPTION
**Pull request by kulbitski-andrei**
pytests for bank_module.py, library_module.py
fixed logging (moved to separate file)
reports added

### CODE REVIEW

File: homework_21/report_21_2.html
Comment (Stml89) [inline homework_21/report_21_2.html:1]: Log report should be in the `.gitignore`

File: homework_21/report_21_1.html
Comment (Stml89) [inline homework_21/report_21_1.html:1]: Log report should be in the `.gitignore`

File: log_dir/log_setup.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import logging
+
+formatter = logging.Formatter(
+    '\n%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+handler = logging.StreamHandler()
+handler.setFormatter(formatter)
+
+logger = logging.getLogger(__name__)
+logger.addHandler(handler)
+logger.setLevel(logging.DEBUG)
```
Comment (Stml89) [inline log_dir/log_setup.py:12]: Requirements says: "Есть возможность запустить тесты с разным уровнем логгирования без изменения кода(те передача уровня логгирования во время запуска тестов)"

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework_21/report_21_2.html
Comment (kulbitski-andrei) [reply homework_21/report_21_2.html:1]: Done

File: homework_21/report_21_1.html
Comment (kulbitski-andrei) [reply homework_21/report_21_1.html:1]: Done

File: log_dir/log_setup.py (hunk 1 above)
Comment (kulbitski-andrei) [reply log_dir/log_setup.py:12]: Not sure I completely understand what's needed. I use command:
pytest --log-cli-level=WARNING <pat_to_file>
and logs are displayed correctly, according to chosen level of logging. Please see attached screenshots:
image
image
image

The logger and handler above just provide "default" level. But this default level is overrided by the level user chooses while running a test with console command.

File: .gitignore [hunk 2]
```diff
@@ -13,3 +15,10 @@ venv/
 #  and can be added to the global gitignore or merged into this file.  For a more nuclear
 #  option (not recommended) you can uncomment the following to ignore the entire idea folder.
 .idea/
+homework_21/assets/style.css
+homework_21/report_21_1.html
+homework_21/report_21_2.html
+
+/output.xml
+/log.html
+/report.html
```
Comment (Stml89) [inline .gitignore:24]: assets/
*.css
*.html
*.xml

File: homework_21/report_21_1.html
Comment (Stml89) [inline homework_21/report_21_1.html:1]: should be in `.gitignore`

File: homework_21/report_21_2.html
Comment (Stml89) [inline homework_21/report_21_2.html:1]: should be in `.gitignore`

File: homework_21/test_homework_21_1.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    return bank_module.Client("test_client_4", -100, bank_module.usd)
+
+
+def test_deposit_with_valid_values(bank_instance, client_1):
+    """Positive test for deposit method"""
+    result = bank_instance.deposit(client_1, 1)
+    assert result == Decimal('110.47')
+    log_setup.logger.info("LOGGER INFO MESSAGE: Test passed")
+    log_setup.logger.debug("LOGGER DEBUG MESSAGE: Test passed")
+    log_setup.logger.warning("LOGGER WARNING MESSAGE: Test passed")
```
Comment (Stml89) [inline homework_21/test_homework_21_1.py:46]: No need to use all log levels, enough INFO
```suggestion
    log_setup.logger.info("LOGGER INFO MESSAGE: Test passed")
```
and fix for other TCs

**Review by Stml89 — APPROVED** [review]

**Pull request merged**